### DISCUSSION TITLE
Question regarding initial dataset

### DISCUSSION THREAD

**Discussion by poyi81** [deepmodeling/dpgen#1409] [Q&A] [answered]
Hi everyone, I’m a rookie to the dpgen. I’m preparing the initial data set for training, and I have a few questions regarding this now.
1. When preparing the dataset, can I directly run AIMD with different temperatures and put the results into the initial dataset? And in the model_deviation step, I will assign the temperature to the corresponding temperature. 
    For example, I have run 100, 200, and 300k AIMD simulations for one structure. Can I set the temperature to 100, 200, and 300k to the corresponding initial structure in the model deviation step? 
   This question seems unreasonable to me, but one of my friends suggested me to do so.

2. In most of the literature, the temperature starts from 50 or 100k to 2 times the melting point. And as the iteration index goes up, the temperature goes up. But in my simulation, the temperature range will be at 300~1000k. To prevent wasting computational resources, can I prepare my initial dataset at 300k and the training starts from 300k and slowly increases to 1000k during iteration index increases?

3. My system will have [Mg,O,H,Al,C]. I have used init_bulk to prepare some of the initial data for this system, but in the `type_map.raw`, it only has [Mg,O,H,C] since this structure has no Al. Can I directly change `type_map.raw` from [Mg,O,H,C] to [Mg,O,H,Al,C] and also change `type.raw` to match the `type_map.raw`?

**Comment by njzjz** [comment, marked as answer]
> * When preparing the dataset, can I directly run AIMD with different temperatures and put the results into the initial dataset? And in the model_deviation step, I will assign the temperature to the corresponding temperature.
>   For example, I have run 100, 200, and 300k AIMD simulations for one structure. Can I set the temperature to 100, 200, and 300k to the corresponding initial structure in the model deviation step?
>   This question seems unreasonable to me, but one of my friends suggested me to do so.

Please note that AIMD usually uses a low-level theory (for example, low convergence criteria) for higher performance. The data used for training should be relabeled using the high-level ab initio theory.

The temperature in the model deviation step is NOT required to match the initial simulations.

> * In most of the literature, the temperature starts from 50 or 100k to 2 times the melting point. And as the iteration index goes up, the temperature goes up. But in my simulation, the temperature range will be at 300~1000k. To prevent wasting computational resources, can I prepare my initial dataset at 300k and the training starts from 300k and slowly increases to 1000k during iteration index increases?

It looks reasonable.

> * My system will have [Mg,O,H,Al,C]. I have used init_bulk to prepare some of the initial data for this system, but in the `type_map.raw`, it only has [Mg,O,H,C] since this structure has no Al. Can I directly change `type_map.raw` from [Mg,O,H,C] to [Mg,O,H,Al,C] and also change `type.raw` to match the `type_map.raw`?

You don't need to change anything.

**Reply by poyi81** [reply]
Regarding the third question, I don't understand what you mean by "You don't need to change anything." I might not specify the question clearly.
Eventually, I will need to train [Mg,O,H,Al,C]. I have already generated some of the initial datasets for this training. But inside this existing initial dataset, the `type_map.raw` is [Mg,O,H,C]. My question is: Can I directly change 'type_map.raw' from [Mg,O,H,C] to [Mg,O,H,Al,C] and also change `type.raw` to match the `type_map.raw`? By doing this, all my initial datasets will have the same `type_map.raw`.
If I remain unchanged, one of the `type_map.raw` would be [Mg,O,H,C] others would be [Mg,O,H,Al,C]. In my understanding, during training, this will end up in error or mismatching of elements.
Thanks for your patience.

**Reply by njzjz** [reply]
Datasets do not need to have the same `type_map.raw`.

**Reply by poyi81** [reply]
>{
     "type_map": ["Mg","O","H","C","Al"],
     "mass_map": [24.305,16,1,12,27],
     "init_data_prefix": "../",
     "init_data_sys": ["init/POSCAR.01x01x01mgoh/02.md/sys-0001-0002-0002/deepmd",
                       "init/POSCAR.01x01x01mgo/02.md/sys-0004-0004/deepmd",
                       "init/2h2o1/2h2o1",
                       "init/2h2o2/2h2o2",
                       "init/2h2o3/2h2o3",
                       "init/2h2o4/2h2o4",

Here is part of my para.json file for running dpgen training. So what you mean is that I don't need all "init_data_sys" to have the same `type_map.raw`. Some of them are [Mg,O,H,Al,C] and some of them are [Mg,O,H,C] can still work as long as `type.raw` matches to `type_map.raw` in each "init_data_sys"? 
Thanks for responding to my naive question.

**Reply by njzjz** [reply]
That's correct.

**Reply by poyi81** [reply]
Thank you very much for responding!

**Reply by 21al07se09t** [reply]
Hi! Could you please clarify the concept of “relabeled” training data？ 
For example, I found the energy of system will be converged by using the ENCUT of 1000, can I prepare the data for init_data_sys by extracting serveral snapshots from AIMD simulations with ENCUT of 400 and set the ENCUT for "fp" step to 1000?
or the "relabeled" means that I should recalculate the single-point energy for the extracted snapshots (from ENCUT=400 AIMD) with ENCUT of 1000 and use them as training data?

**Reply by njzjz** [reply]
@21al07se09t The second one is correct.

**Reply by 21al07se09t** [reply]
Thank you so much!
